yarpl/test/Observable_test.cpp

```diff
@@ -386,6 +386,16 @@ TEST(Observable, DISABLED_SimpleTake) {
   EXPECT_EQ(0u, Refcounted::objects());
 }
 
+TEST(Observable, IgnoreElements) {
+  ASSERT_EQ(0u, Refcounted::objects());
+
+  EXPECT_EQ(
+      run(Observables::range(0, 100)->ignoreElements()),
```

Comment:
Run method makes the consumer to consume 100 elements. Please check even if we consume less than 100 elements, ignoreElements function completes. Namely change the range from 0, 100 to 0, 200 and/or 0, 50 see what happens.